.idea/Working version/algorithms.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * Created by tsodergren on 6/5/17.
+ */
+
+
+function constructEdges() {
+
+
+    var radiusChange = arguments.length ? arguments[0] : 0;
+
+    //squared coverage diameter and annulus of uncertainty
+    var sqDiameter = 4 * Math.pow(complexRadius, 2);
+    var sqDiameterMin = 4 * Math.pow(complexRadius-dataRadius, 2);
+    var sqDiameterMax = 4 * Math.pow(complexRadius+dataRadius, 2);
+
+
+    if (!a_edges) {
+        var x=math.zeros(numSamples, numSamples,'sparse');
+        a_edges = {edgeProb: x.clone(), edgeAnchorDist: x.clone(), edgeDist: x.clone(), edgeFlag: x.clone(),
+            edgeInd: x.clone() };
+    }
+
+    var iEdgeDist, iEdgeFlag, tempEdges;
+
+    if ( radiusChange==1 ) {
+        tempEdges = JSON.parse(JSON.stringify(ripsEdges));
```

Comment:
This change was actually made to the main branch before I started working on the algorithm and done to make a copy of the array. Originally I was using array.slice(), but while it makes a shallow copy of the array, with object arrays it just copies the object references, so any changes affect the original array. This is what was causing that issue earlier with the Rips probabilities getting overwritten by the Cech probabilities. After some research, this seems to be the cleanest way of making a deep copy and leaving the original array intact.